app.rb

```diff
@@ -0,0 +1,119 @@
+require_relative 'book'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rental'
+require_relative 'person'
+
+class App
+  attr_accessor :books, :person, :rentals
+
+  def initialize
+    @books = []
+    @persons = []
+    @rentals = []
+  end
+
+  def list_all_books
+    puts "\nList of Books. \n"
+    @books.each do |book|
+      puts "title: #{book.title} by author : #{book.author}"
+    end
+    puts "\n"
+  end
+
+  def list_all_people
+    puts "\n"
+    @persons.each do |person|
+      if person.instance_of?(Teacher)
+        puts "[Teacher]Age: #{person.age} Name: #{person.name}, ID: #
+                {person.id}"
+      else
+        puts "[Student]Age: #{person.age},Name: #{person.name}, ID: #
+                {person.id}"
+      end
+    end
+    puts "\n"
+  end
+
+  def create_a_person
+    print "\nDo you want to add a student (1) or a teacher (2)? [Insert the number]: "
+    is_student = gets.chomp.strip.to_i
+    print 'age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    case is_student
+    when 1
+      print 'Has Parent Permission [Y/N]: '
+      permission = gets.chomp.downcase == 'y'
+      create_student(age, name, permission)
+    when 2
+      create_teacher(age, name)
+    end
+    print "Person added Successfully. \n"
+    $stdout.flush
+  end
```

Comment:
- [ ] Great job working on the requested changes but I think it would be great if you fix the issue that the previous reviewer mentioned since when I try to create a person I face this error so please fix the issue to be able to create a person I face this error when I try to create both student or teacher ⬇♻

- When I create student
![image](https://github.com/brighton0102/School-Library-App/assets/58553711/d431bc0b-3f38-4130-bc32-6520634a68c9)

- when I create a teacher
![image](https://github.com/brighton0102/School-Library-App/assets/58553711/fd825741-57a8-4008-8e40-44159269884c)